scripts/prove_rpc.sh

```diff
@@ -108,49 +93,76 @@ fi
+--rpc-url $NODE_RPC_URL \
+--start-block $START_BLOCK \
+--end-block $END_BLOCK \
+--backoff $BACKOFF \
+--max-retries $RETRIES \
+$PREV_PROOF_EXTRA_ARG"
+
     if [ "$OUTPUT_TO_TERMINAL" = true ]; then
-        eval $command
+        eval "$command"
         retVal=$?
         echo -e "Proof witness generation finished with result: $retVal"
         exit $retVal
     else
-        eval $command > $OUT_LOG_PATH 2>&1
-        if grep -q 'All proof witnesses have been generated successfully.' $OUT_LOG_PATH; then
+        eval "$command" > "$OUT_LOG_PATH" 2>&1
+        if grep -q 'All proof witnesses have been generated successfully.' "$OUT_LOG_PATH"; then
             echo -e "Success - Note this was just a test, not a proof"
             # Remove the log on success if we don't want to keep it.
             if [ $ALWAYS_WRITE_LOGS -ne 1 ]; then
-                rm $OUT_LOG_PATH
+                rm "$OUT_LOG_PATH"
             fi
             exit
         else
-            echo "Failed to create proof witnesses. See ${OUT_LOG_PATH} for more details."
+            echo "Failed to create proof witnesses. See $OUT_LOG_PATH for more details."
             exit 1
         fi
     fi
 else
     # normal run
-    echo "Proving blocks ${BLOCK_INTERVAL} now... (Total: ${TOT_BLOCKS})"
-    command='cargo r --release --package zero --bin leader -- --runtime in-memory --load-strategy on-demand --proof-output-dir $PROOF_OUTPUT_DIR --block-batch-size $BLOCK_BATCH_SIZE rpc --rpc-type "$NODE_RPC_TYPE" --rpc-url "$3" --block-interval $BLOCK_INTERVAL $PREV_PROOF_EXTRA_ARG --backoff "$BACKOFF" --max-retries "$RETRIES" '
+    echo "Proving blocks from ($START_BLOCK) to ($END_BLOCK)"
+    command="cargo r --release --package zero --bin leader -- \
+--runtime in-memory \
+--load-strategy on-demand \
+--proof-output-dir $PROOF_OUTPUT_DIR \
+--block-batch-size $BLOCK_BATCH_SIZE \
+rpc \
+--rpc-type $NODE_RPC_TYPE \
+--rpc-url $3 \
+--start-block $START_BLOCK \
+--end-block $END_BLOCK \
+--backoff $BACKOFF \
+--max-retries $RETRIES \
+$PREV_PROOF_EXTRA_ARG "
+
     if [ "$OUTPUT_TO_TERMINAL" = true ]; then
-        eval $command
+        eval "$command"
         echo -e "Proof generation finished with result: $?"
     else
-        eval $command > $OUT_LOG_PATH 2>&1
+        eval "$command" > "$OUT_LOG_PATH" 2>&1
         retVal=$?
         if [ $retVal -ne 0 ]; then
             # Some error occurred, display the logs and exit.
-            cat $OUT_LOG_PATH
-            echo "Block ${i} errored. See ${OUT_LOG_PATH} for more details."
+            cat "$OUT_LOG_PATH"
+            echo "Error occurred. See $OUT_LOG_PATH for more details."
             exit $retVal
         else
             # Remove the log on success if we don't want to keep it.
             if [ $ALWAYS_WRITE_LOGS -ne 1 ]; then
-                rm $OUT_LOG_PATH
+                rm "$OUT_LOG_PATH"
             fi
         fi
-        echo "Successfully generated ${TOT_BLOCKS} proofs!"
```

Comment:
Maybe you could count the block numbers by grepping and counting successful block log info lines in the log file?

Reply:
I have grep'd for a string based on what I could see in the output with `test_only`. LMK if it should be something else.
```
INFO zero::prover: Proving block \d'
```